File: launchmode/src/main/java/com/example/launchmode/MainActivity.java
package com.example.launchmode;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import java.net.Inet4Address;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void next(View view) {
        Log.d(TAG, "------------- start activity -------------");
        Intent intent = new Intent(this, SingleInstanceActivity.class);
        startActivity(intent);
    }
}

File: daggerdemo/src/main/java/com/example/daggerdemo/MainActivity.kt
package com.example.daggerdemo

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import com.example.daggerdemo.base.MainContract
import dagger.android.support.DaggerAppCompatActivity
import javax.inject.Inject

class MainActivity : DaggerAppCompatActivity(), MainContract.IView {

    lateinit var text: TextView

    @Inject
    lateinit var presenter: MainContract.IPresenter

    companion object {
        private const val TAG = "MainActivity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        text = findViewById(R.id.dagger_text)
        presenter.takeView(this)
        text.setOnClickListener {
            val intent = Intent(MainActivity@ this, SecondActivity::class.java)
            startActivity(intent)
        }
    }


    override fun onResume() {
        super.onResume()
        presenter.getData()
    }

    override fun showText(content: String) {
        text.setText(content)
    }
}